Ska/Matplotlib/core.py

```diff
@@ -142,14 +142,13 @@ def plot_cxctime(times, y, fmt=None, fig=None, ax=None, yerr=None, xerr=None, tz
         if fmt is not None:
             kwargs.update({'fmt': fmt})
         ax.errorbar(cxctime2plotdate(times), y, yerr=yerr, xerr=xerr, **kwargs)
-        ax.xaxis_date(tz)
     else:
+        if fmt is not None:
+            ax.plot(CxoTime(times).datetime, y, fmt, **kwargs)
```

Comment:
Problem here is that the `datetime` format generates an object array of `datetime.datetime` objects, which is going to be unhappy for a large array of times. 

There is a simple fix though, use `CxoTime(times).datetime64` instead which makes an array of numpy datetime64. This will be fast and efficient.

I think that the format conversion can be put up front as below and used in `errorbar` and `plot`.
```
times = CxoTime(times).datetime64
```
With this preprocessing I found in testing that the `ax.xaxis_date` is not longer needed.

Reply:
> With this preprocessing I found in testing that the ax.xaxis_date is not longer needed.

This also makes it incompatible with previous versions of matplotlib. It might not matter though.

Reply:
Agreed, given how we operate and deliver ska3 as an integrated environment, there is no problem with incompatibility with previous versions of matplotlib.

Reply:
If you put the format conversion up front for `errorbar` and `plot` it looks like xerr doesn't behave.  

```
plot_cxctime(t, y, '--g', yerr=5, xerr=.5);
-------------------------------------------------------------------------
UFuncTypeError                            Traceback (most recent call last)
/tmp/ipykernel_7334/3216467896.py in <module>
----> 1 plot_cxctime(t, y, '--g', yerr=5, xerr=.5);

~/git/Ska.Matplotlib/Ska/Matplotlib/core.py in plot_cxctime(times, y, fmt, fig, ax, yerr, xerr, state_codes, interactive, **kwargs)
    143         if fmt is not None:
    144             kwargs.update({'fmt': fmt})
--> 145         ax.errorbar(times, y, yerr=yerr, xerr=xerr, **kwargs)
    146     else:
    147         if fmt is not None:

/fido.real/conda/envs/ska3-next/lib/python3.8/site-packages/matplotlib/__init__.py in inner(ax, data, *args, **kwargs)
   1410     def inner(ax, *args, data=None, **kwargs):
   1411         if data is None:
-> 1412             return func(ax, *map(sanitize_sequence, args), **kwargs)
   1413 
   1414         bound = new_sig.bind(ax, *args, **kwargs)

/fido.real/conda/envs/ska3-next/lib/python3.8/site-packages/matplotlib/axes/_axes.py in errorbar(self, x, y, yerr, xerr, fmt, ecolor, elinewidth, capsize, barsabove, lolims, uplims, xlolims, xuplims, errorevery, capthick, **kwargs)
   3463             #     return dep - elow * ~lolims, dep + ehigh * ~uplims
   3464             # except that broadcast_to would strip units.
-> 3465             low, high = dep + np.row_stack([-(1 - lolims), 1 - uplims]) * err
   3466 
   3467             barcols.append(lines_func(

UFuncTypeError: ufunc 'add' cannot use operands with types dtype('<M8[ns]') and dtype('O')
```

Reply:
I rolled it back to use cxctime2plotdate .